main/src/io/github/iltotore/iron/RuntimeConstraint.scala

```diff
@@ -0,0 +1,21 @@
+package io.github.iltotore.iron
+
+import scala.util.NotGiven
+
+final class RuntimeConstraint[A, T](_test: A => Boolean, val message: String):
+  inline def test(value: A): Boolean = _test(value)
+
+object RuntimeConstraint:
+
+  inline def derived[A, C](using inline c: Constraint[A, C]): RuntimeConstraint[A, A :| C] =
+    new RuntimeConstraint[A, A :| C](c.test(_), c.message)
+
+  final class AutoDerived[A, T](val inner: RuntimeConstraint[A, T])
+  object AutoDerived:
+    inline given [A, C](using rtc: RuntimeConstraint[A, A :| C]): AutoDerived[A, A :| C] =
+      new AutoDerived[A, A :| C](rtc)
+
+    inline given [A, C](using inline c: Constraint[A, C], ng: NotGiven[RuntimeConstraint[A, A :| C]]): AutoDerived[A, A :| C] =
+      new AutoDerived[A, A :| C](RuntimeConstraint.derived[A, C])
```

Comment:
One of the things that was bugging me was having auto-derivation of `RuntimeConstraint` from a `Constraint` enabled by default. This is fairly dangerous as methods requiring a `RuntimeConstraint` would auto-derive one if one wasn't in scope, which can lead to increased compilation time and generated code size.

With this, a user / method / etc can define whether they want an explicit or implicit given to be provided. For newtypes, this is generated by default

Example:

```scala
// will raise a compile-time error if one isn't explicitely defined (newtypes are OK since we create one for them in RefinedTypeOpsImpl)
def foo(value: Int)(using RuntimeConstraint[Int, Int :| Greater[10]]) = ??? 

 // Will prioritise an explicitly defined RuntimeConstraint and auto-derive one if it doesn't exist
def foo(value: Int)(using RuntimeConstraint.AutoDerived[Int, Int :| Greater[10]]) = ???
```

Reply:
I still don't understand why `RuntimeConstraint` does not have the same type parameters than `Constraint`. Then, it would look like:

```scala
def foo(value: Int)(using RuntimeConstraint[Int, Greater[10]]) = ???

def foo(value: Int)(using RuntimeConstraint.AutoDerived[Int, Greater[10]]) = ???
```

Also, I'm not sure about the usefulness of `AutoDerived`. It would leak an implementation detail (how the instance is generated) and I'm not sure about the real impact of auto deriving automatically on bytecode size and compilation times.

Reply:
After sleeping on it, I think you're right. I reverted these changes